README.md

```diff
@@ -131,15 +141,397 @@ class Admin < Dill::Role
 end
 ```
 
-##To Be Continued...
+That's it for the walkthrough, but Dill has a lot more to offer. We recommend digging into Widgets next, as Widget is the fundamental class common to all the Dill elements.
+
+
+#Widgets
+A widget is the fundamental Dill element. A widget abstracts a DOM element, allowing you to call methods on that element, like checking for text or submitting a form.
+
+
+## Widget Declaration
+A widget can be declared with the `widget` macro, or as a class.
+
+Macro widget declarations take a CSS or XPath selector as the first argument.
+
+```ruby
+  widget :todo_item, '.todo-item'
+  widget :todo_item, [:xpath, '//some/node']
+```
+
+Class widget declarations define the widget with selector via the `root` macro.
+
+```ruby
+class TodoItem < Dill::Widget
+  root '.todo-item'
+end
+
+class TodoItem < Dill::Widget
+  root :xpath, '//some/node'
+end
+```
+
+## Nested Widgets
+Widgets can be defined within widgets. CSS classes of child widgets are scoped within the class of the parent widget.
+
+Child widgets are available via the `widget` method.
+
+```ruby
+class TodoManager < Dill::Role
+  widget :todo_item, '.todo-item' do
+    widget :delete_link, '.delete' # :delete_link looks for a .delete class within a .todo-item element
+  end
+
+  # to access the delete_link widget:
+  # widget(:todo_item).widget(:delete_link)
+end
+```
+
+
+## Widget Methods
+You can define custom methods on a widget object, or use the built-in methods.
+
+```ruby
+class TodoManager < Dill::Role
+  widget :todo_item, '.todo-item' do
+    widget :delete_link, '.delete'
+
+    def delete
+      click :delete_link
+    end
+  end
+
+  def delete_item
+    widget(:todo_item).delete
+  end
+
+  def see_deletable_todo_item?
+    # check for widget presence with the widget? method
+    widget(:todo_item).widget?(:delete_link)
+  end
+end
+```
+
+These methods can be called like this:
+
+```ruby
+roles.todo_manager.delete_item
+
+expect(roles.todo_manager).to see :deletable_todo_item
+```
+
+
+## Widgets As Procs
+Widgets can take a Proc iin addition to a CSS selector or XPath expression, allowing us to more precisely define the widget at call time.
+
+```ruby
+class TodoManager < Dill::Role
+  widget :todo_item, -> (text) { ['.todo-item', text: text] } do
+    widget :delete_link, '.delete'
+
+    def delete
+      click :delete_link
+    end
+  end
```

Comment:
This is where what I wrote above breaks down a bit, unless you're willing to use xpath. Otherwise, you'd have to pass in an `id` instead of `text`:

```ruby
widget :delete_todo_item, -> (id) { "#todo_item_#{id} .delete" }

def delete(todo_item)
  click :delete_todo_item, todo_item.id
end
```